index.js

```diff
@@ -0,0 +1,247 @@
+const express = require("express");
+const xrpl = require("xrpl");
+require("dotenv").config();
+const { Attendify } = require("./attendify");
+const {
+  postToIPFS,
+  ascii_to_hexa,
+  ERR_ATTENDIFY,
+  ERR_IPFS,
+  ERR_NOT_FOUND,
+  ERR_PARAMS,
+  ERR_XRPL,
+} = require("./utils");
+const fs = require("fs");
+
+const app = express();
+let AttendifyLib = new Attendify();
+
+/**
+ * Creates new event,
+ * Uploads its metadata to IPFS
+ * and mints a batch of NFTs for it
+ * @route GET /api/mint
+ * @param {string} walletAddress - The wallet address to mint the NFT tokens to
+ * @param {integer} tokenCount - The number of NFT tokens to mint
+ * @param {string} url - The URL of the event
+ * @param {string} title - The title of the event
+ * @param {string} desc - The description of the event
+ * @param {string} loc - The location of the event
+ * @returns {object} result - An object with the mint result
+ * @throws {Error} If any of the walletAddress, tokenCount, url, title, desc, or loc parameters are missing or have an invalid value
+ */
+app.get("/api/mint", (req, res) => {
+  (async () => {
+    try {
+      const { walletAddress, tokenCount, url, title, desc, loc } =
+        await req.query;
+      if (
+        walletAddress.length == 0 ||
+        tokenCount.length == 0 ||
+        url.length == 0 ||
+        title.length == 0 ||
+        walletAddress.length == 0 ||
+        desc.length == 0 ||
+        loc.length == 0
+      )
+        throw new Error(`${ERR_PARAMS}`);
+      const vaultWallet = await xrpl.Wallet.fromSeed(process.env.WALLET_SEED);
+      let metadataStructure = {
+        title: title,
+        description: desc,
+        collectionSize: tokenCount,
+        location: loc,
+        date: new Date().toLocaleDateString().toString(),
+        URI: url,
+        account: vaultWallet.address,
+      };
+
+      const metadata = await postToIPFS(JSON.stringify(metadataStructure)); //.substring(21);
+
+      console.log(metadata);
+
+      return res.send({
+        result: await AttendifyLib.batchMint(
```

Comment:
While batchMint does now use Tickets, it seems to still be doing one mint per ledger as opposed to leveraging the tickets to do a couple at the same time (Screenshot attached) - For small numbers it's not a big deal, but it could lead to long wait times as you start minting 1000+ NFTs. ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/eed79aa5-104f-4148-9fbb-26439fa30bcd/Untitled.png)

Reply:
(Here I minted 5 NFTs using api/mint, and they were actually minted about once per ledger)

Reply:
That was caused by old code that used  `submitAndWait` instead of `submit`. I have already changed it and from what I've seen in explorer multiple NFTs are minted at the same time now.